crates/iota-indexer/reliability.md

```diff
@@ -0,0 +1,54 @@
+| `latest_fullnode_checkpoint_sequence_number`             | Latest checkpoint from full node ready to be synced by the indexer.                                                 |
+| `max_committed_checkpoint_sequence_number`               | The highest committed checkpoint by the indexer.                                                                    |
+| `checkpoint_db_commit_latency`                           | Total latency to commit a chunk of checkpoints (including all transactions, objects, events, etc.) to the database. |
+| `checkpoint_db_commit_latency_transactions_chunks`       | Latency to commit a single chunk of checkpoint transactions to the database.                                        |
+| `checkpoint_db_commit_latency_tx_insertion_order_chunks` | Latency to commit as single chunk of data representing the transaction insertion order to the database.             |
+| `checkpoint_db_commit_latency_tx_indices_chunks`         | Latency to commit a single chunk of data representing the transaction indices to the database.                      |
+| `checkpoint_db_commit_latency_events_chunks`             | Latency to commit a single chunk of data representing the events to the database.                                   |
+| `checkpoint_db_commit_latency_event_indices_chunks`      | Latency to commit a single chunk of data representing the events indices to the database.                           |
+| `checkpoint_db_commit_latency_packages`                  | Latency to commit a single chunk of data representing the packages to the database.                                 |
+| `checkpoint_db_commit_latency_objects_chunks`            | Latency to commit a single chunk of data representing the objects to the database.                                  |
+| `checkpoint_db_commit_latency_objects_history_chunks`    | Latency to commit a single chunk of data representing the object history to the database.                           |
+| `checkpoint_db_commit_latency_objects_snapshot_chunks`   | Latency to commit a single chunk of data representing the object snapshot to the database.                          |
+| `checkpoint_db_commit_latency_objects_snapshot`          | Total latency to commit a batch of checkpoints object snapshots to the database.                                    |
+| `checkpoint_db_commit_latency_epoch`                     | Latency to commit the epoch data into database.                                                                     |
+| `req_latency_by_route`                                   | Latency of indexer JSON RPC API endpoints. Measures the time from request receipt to response delivery.             |
+
+> [!Note]
+>
+> - Large datasets (e.g., transactions, objects) are split into smaller **chunks** to avoid overwhelming the database with bulk writes.
```

Comment:
What is the default chunk size?

Reply:
fixed: https://github.com/iotaledger/iota/pull/6694/commits/96fa180bfbd0c6d8db38c51a06afacd9b43ac2b9

- The primary pipeline uses by default chunks of **100** elements, it can be modified by this env var `PG_COMMIT_PARALLEL_CHUNK_SIZE`.
- The object snapshot pipeline uses by default chunks of **500** elements, it can be modified by this env var `PG_COMMIT_OBJECTS_PARALLEL_CHUNK_SIZE`.